addons/barcodes_gs1_nomenclature/static/src/js/epc_model.js

```diff
@@ -0,0 +1,279 @@
+    },
+    0x32: {
+        name: "SGLN-96",
+        length: 24,
+        uri_pattern_tag: "urn:epc:tag:sgln-96:F.C.L.E",
+        uri_pure_tag: "urn:epc:id:sgln:C.L.E",
+        partition: "sgln",
+        fields_template: [
+            "header",
+            "filter",
+            "partition",
+            "company_prefix",
+            "location_reference",
+            "extension_integer",
+        ],
+        fields_bit_count: [8, 3, 3, 0, 0, 41],
+        ai_list: {
+            "414": null,
+            "254": "extension_integer",
+        },
+    },
+    0x39: {
+        name: "SGLN-195",
+        length: 52,
+        uri_pattern_tag: "urn:epc:tag:sgln-195:F.C.L.E",
+        uri_pure_tag: "urn:epc:id:sgln:C.L.E",
+        partition: "sgln",
+        fields_template: [
+            "header",
+            "filter",
+            "partition",
+            "company_prefix",
+            "location_reference",
+            "extension_string",
+        ],
+        fields_bit_count: [8, 3, 3, 0, 0, 140],
+        ai_list: {
+            "414": null,
+            "254": "extension_string",
+        },
+    },
+};
+
+export const FIELD_TEMPLATE = {
+    header: {
+        name: "EPC Header",
+        encoding: "integer",
+    },
+    filter: {
+        name: "Filter",
+        encoding: "integer",
+        uri_portion: "F",
+    },
+    partition: {
+        name: "Partition",
+        encoding: "partition_table",
+    },
+    company_prefix: {
+        name: "Company Prefix",
+        encoding: "integer",
+        uri_portion: "C",
+    },
+    item_reference: {
+        name: "Item Reference",
+        encoding: "integer",
+        uri_portion: "I",
+    },
+    serial_integer: {
+        name: "Serial (Integer)",
+        encoding: "integer",
+        uri_portion: "S",
+    },
+    serial_string: {
+        name: "Serial (Alphanumeric)",
+        encoding: "string",
+        uri_portion: "S",
+    },
+    blank: {
+        name: "Blank Filler",
+        encoding: "blank",
+    },
+    location_reference: {
+        name: "Location Reference",
+        encoding: "integer",
+        uri_portion: "L",
+    },
+    extension_integer: {
+        name: "Extension (Integer)",
+        encoding: "integer",
+        uri_portion: "E",
+    },
+    extension_string: {
+        name: "Extension (Alphanumeric)",
+        encoding: "string",
+        uri_portion: "E",
+    },
+};
+
+export const PARTITION = {
+    sgtin: {
+        0: { left_bit: 40, left_digit: 12, right_bit: 4, right_digit: 1 },
+        1: { left_bit: 37, left_digit: 11, right_bit: 7, right_digit: 2 },
+        2: { left_bit: 34, left_digit: 10, right_bit: 10, right_digit: 3 },
+        3: { left_bit: 30, left_digit: 9, right_bit: 14, right_digit: 4 },
+        4: { left_bit: 27, left_digit: 8, right_bit: 17, right_digit: 5 },
+        5: { left_bit: 24, left_digit: 7, right_bit: 20, right_digit: 6 },
+        6: { left_bit: 20, left_digit: 6, right_bit: 24, right_digit: 7 },
+    },
```

Comment:
If keys go always from 0 to 6, why not simply define them as an array instead ?
```suggestion
    sgtin: [
        { left_bit: 40, left_digit: 12, right_bit: 4, right_digit: 1 },
        { left_bit: 37, left_digit: 11, right_bit: 7, right_digit: 2 },
        { left_bit: 34, left_digit: 10, right_bit: 10, right_digit: 3 },
        { left_bit: 30, left_digit: 9, right_bit: 14, right_digit: 4 },
        { left_bit: 27, left_digit: 8, right_bit: 17, right_digit: 5 },
        { left_bit: 24, left_digit: 7, right_bit: 20, right_digit: 6 },
        { left_bit: 20, left_digit: 6, right_bit: 24, right_digit: 7 },
    ],
```
If it's for clarity purpose, it's totally right to keep it as now